SoundSwitch/UI/Forms/Settings.cs

```diff
@@ -868,60 +849,100 @@ private void ResetAudioDevicesButton_Click(object sender, EventArgs e)
                 DefaultExt = "zip",
                 Filter = "Zip Archive (*.zip)|*.zip",
-                RestoreDirectory = true
+                RestoreDirectory = true,
             };
 
             if (saveFileDialog.ShowDialog() != DialogResult.OK) return;
             if (File.Exists(saveFileDialog.FileName))
                 File.Delete(saveFileDialog.FileName);
 
-            Log.CloseAndFlush();
+            using var archive = ZipFile.Open(saveFileDialog.FileName, ZipArchiveMode.Create);
+            exportArchive(archive);
+        }
 
-            var files = Directory.EnumerateFiles(ApplicationPath.Logs, "*.log");
-            using var zip = ZipFile.Open(saveFileDialog.FileName, ZipArchiveMode.Create);
-            foreach (var file in files)
+        private void ExportLogFilesButton_Click(object sender, EventArgs e)
+        {
+            PrepareZipArchive(SettingsStrings.exportLogFiles, "soundswitch_logs", archive =>
             {
-                // Add the entry for each file
-                zip.CreateEntryFromFile(file, Path.GetFileName(file), CompressionLevel.Optimal);
-            }
+                Log.CloseAndFlush();
 
-            LoggerConfigurator.ConfigureLogger();
-        }
+                var files = Directory.EnumerateFiles(ApplicationPath.Logs, "*.log");
+                foreach (var file in files)
+                    // Add the entry for each file
+                    archive.CreateEntryFromFile(file, Path.GetFileName(file), CompressionLevel.Optimal);
 
-        private void GitHubHelpLinkLabel_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
-        {
-            GitHubHelpLink();
+                LoggerConfigurator.ConfigureLogger();
+            });
         }
 
-        private void DiscordCommunityLinkLabel_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
+        private void ExportConfigFileButton_Click(object sender, EventArgs e)
         {
-            DiscordCommunityLink();
+            PrepareZipArchive(SettingsStrings.exportConfigFile, "soundswitch_config", archive =>
+            {
+                const string configFile = "SoundSwitchConfiguration.json";
+                archive.CreateEntryFromFile(Path.Combine(ApplicationPath.Default, configFile), configFile, CompressionLevel.Optimal);
+            });
         }
 
-        private void DonateLinkLabel_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
+        private void ImportConfigFileButton_Click(object sender, EventArgs e)
         {
-            DonateLink();
+            const string configFile = "SoundSwitchConfiguration.json";
+            var filePath = Path.Combine(ApplicationPath.Default, configFile);
+
+            var openFileDialog = new OpenFileDialog
+            {
+                Title = SettingsStrings.importConfigFile,
+                Filter = "Zip Archive (*.zip)|*.zip",
+                RestoreDirectory = true,
+            };
+
+            while (true)
```

Comment:
Why the while (true) ?

Shouldn't the logic simply be: Check given that the zip has the wanted file, then return either error or success with restart application ?